### PR TITLE
Refactor the code validateRequiredFinancialServiceProviderConfiguration

### PR DESCRIPTION
**Pull request by RubenGeo**
[AB#30488](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/30488) 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests wherever relevant
- [ ] I have made sure that all automated checks pass before requesting a review
- [ ] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts [hunk 1]
```diff
@@ -293,7 +293,6 @@ describe('Do failing payment with FSP Visa Debit', () => {
     );
   });
 
-  // ##TODO: Re-enable this test when the new API for FSP configuration is implemented
   it('should fail pay-out by visa debit if coverletterCode is not configured for the program', async () => {
```
Comment (jannisvisser) [inline services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts:296]: add 'or fundingTokenCode' in description

File: services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts [hunk 2]
```diff
@@ -323,6 +329,13 @@ describe('Do failing payment with FSP Visa Debit', () => {
     );
 
     expect(doPaymentResponse.status).toBe(HttpStatus.BAD_REQUEST);
+    // Check if both properties are mentioned in the error message
```
Comment (jannisvisser) [inline services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts:332]: the test fails. I think you should change add JSON.stringify  this to 

```
expect(JSON.stringify(doPaymentResponse.body)).toContain(
      FinancialServiceProviderConfigurationProperties.coverLetterCode,
    );
```

File: services/121-service/src/payments/payments.service.ts [hunk 3]
```diff
@@ -319,10 +317,26 @@ export class PaymentsService {
     return bulkActionResultPaymentDto;
   }
 
-  // ##TODO needs to be refactored to work with programFspConfigProperties
-  async validateRequiredFinancialServiceProviderConfigurations(
+  private async checkFspConfigurationsOrThrow(
```
Comment (jannisvisser) [inline services/121-service/src/payments/payments.service.ts:320]: I tested this functionally by dropping columnToMatch for gringotts and get a generic error in the front-end, correct error in back-end. Is that good enough?

image

**Review by jannisvisser — CHANGES_REQUESTED** [review]

File: services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts (hunk 2 above)
Comment (RubenGeo) [reply services/121-service/test/payment/do-payment-fsp-visa-debit/do-payment-fsp-visa-debit-error.test.ts:332]: I added .message instead

File: interfaces/Portal/src/app/program/make-payment/make-payment.component.ts [hunk 4]
```diff
@@ -194,8 +194,12 @@ export class MakePaymentComponent implements OnInit, OnDestroy {
   }
 
   private onPaymentError(error) {
-    if (error && error.error && error.error.errors) {
-      actionResult(this.alertController, this.translate, error.error.errors);
+    if (error && error.error && (error.error.errors || error.error.message)) {
```
Comment (RubenGeo) [inline interfaces/Portal/src/app/program/make-payment/make-payment.component.ts:197]: I added this to ensure the error message is visible in the portal. The code is not pretty as I don't really know when `error.error.errors` occurs. Do you think it's worth the time to dive into this further?  @jannisvisser

**Review by jannisvisser — APPROVED** [review]

**Pull request merged**